util/util.go

```diff
@@ -424,6 +424,20 @@ func ActivateLibStorage(
 		errs <-chan error
 	)
 
+	// generate cert/key pair if necessary
+	certPath := config.GetString(apitypes.ConfigTLSCertFile)
+	keyPath := config.GetString(apitypes.ConfigTLSKeyFile)
+	host := config.GetString(apitypes.ConfigHost)
+
+	ctx.WithFields(log.Fields{
+		"host":     host,
+		"certPath": certPath,
+		"keyPath":  keyPath,
+	}).Debug("generating self-cert")
+	if err := CreateSelfCert(ctx, certPath, keyPath, host); err != nil {
```

Comment:
Hi @vladimirvivien,

The host from the config file should not be the host used when generating a cert. The config file property isn't in the correct format for a host name used as a TLS certificate's common name (CN) part of the Subject.

My suggestion is to hardcode this to `127.0.0.1` or let user pick the value via a CLI flag. @cduchesne / @codenrhoden, thoughts?